ui/webui/src/apis/storage.js

```diff
@@ -0,0 +1,168 @@
+    return storageClient(address).call(
+        "/org/fedoraproject/Anaconda/Modules/Storage",
+        "org.fedoraproject.Anaconda.Modules.Storage",
+        "CreatePartitioning", [method]
+    );
+};
+
+/**
+ * @param {string} address      Anaconda bus address
+ *
+ * @returns {Promise}           Resolves all properties of DiskSelection interface
+ */
+export const getAllDiskSelection = ({ address }) => {
+    return storageClient(address).call(
+        "/org/fedoraproject/Anaconda/Modules/Storage/DiskSelection",
+        "org.freedesktop.DBus.Properties",
+        "GetAll",
+        ["org.fedoraproject.Anaconda.Modules.Storage.DiskSelection"],
+    );
+};
+
+/**
+ * @param {string} address      Anaconda bus address
+ * @param {string} disk         A device name
+ *
+ * @returns {Promise}           Resolves an object with the device data
+ */
+export const getDeviceData = ({ address, disk }) => {
+    return storageClient(address).call(
+        "/org/fedoraproject/Anaconda/Modules/Storage/DeviceTree",
+        "org.fedoraproject.Anaconda.Modules.Storage.DeviceTree.Viewer",
+        "GetDeviceData", [disk]
+    );
+};
+
+/**
+ * @param {string} address      Anaconda bus address
+ *
+ * @returns {Promise}           Resolves a list with disk names
+ */
+export const getUsableDisks = ({ address }) => {
+    return storageClient(address).call(
+        "/org/fedoraproject/Anaconda/Modules/Storage/DiskSelection",
+        "org.fedoraproject.Anaconda.Modules.Storage.DiskSelection",
+        "GetUsableDisks", []
+    );
+};
+
+/**
+ * @param {string} address      Anaconda bus address
+ * @param {string} partitioning DBus path to a partitioning
+ *
+ * @returns {Promise}           Resolves a DBus path to a task
+ */
+export const partitioningConfigureWithTask = ({ address, partitioning }) => {
+    return storageClient(address).call(
+        partitioning,
+        "org.fedoraproject.Anaconda.Modules.Storage.Partitioning",
+        "ConfigureWithTask", []
+    );
+};
+
+/**
+ * @param {string} address      Anaconda bus address
+ * @param {string} task         DBus path to a task
+ * @param {string} onSuccess    Callback to run after Succeeded signal is received
+ * @param {string} onFail       Callback to run as an error handler
+ *
+ * @returns {Promise}           Resolves a DBus path to a task
+ */
+export const runStorageTask = ({ address, task, onSuccess, onFail }) => {
+    const taskProxy = storageClient(address).proxy(
+        "org.fedoraproject.Anaconda.Task",
+        task
+    );
+    const addEventListeners = () => {
+        taskProxy.addEventListener("Stopped", () => taskProxy.Finish().catch(onFail));
+        taskProxy.addEventListener("Succeeded", onSuccess);
+    };
+    taskProxy.wait(() => {
+        addEventListeners();
+        taskProxy.Start().catch(onFail);
+    });
+};
+
+/**
+ * @param {string} address      Anaconda bus address
+ * @param {int} mode            The number of the mode
```

Comment:
I think it would be good to either enumerate the modes here or mention where they are defined.

Reply:
I really don't want to duplicate the API documentation.
If the documentation for the pyanaconda DBUS api was actually generated and online somewhere, I would gladly add a link to the original method that we expose here. But till then I feel it's the maximum amount of info we should add in this docstring without duplicating docs.

Reply:
OK, makes sense. :)